compiler-rt/lib/builtins/aarch64/sme-libc-mem-routines.S

```diff
@@ -238,6 +236,7 @@ END_COMPILERRT_OUTLINE_FUNCTION(__arm_sc_memcpy)
 
 DEFINE_COMPILERRT_FUNCTION_ALIAS(__arm_sc_memmove, __arm_sc_memcpy)
 
+#if defined(__aarch64__) && __ARM_FP != 0
```

Comment:
This could do with a comment explaining why only some are built when `__ARM_FP != 0`, and pointing to the C versions.

Reply:
Good plan. I've stuck a comment in both .c and .S files about why this function is implemented in both files.